.config/ast-grep/rules/add_function_annotations_to_traits.yml

```diff
@@ -0,0 +1,31 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/ast-grep/ast-grep/main/schemas/rule.json
```

Comment:
Committing this file will cause github actions to run it as a lint (we have an ast-grep job). Thanks for sharing the ast-grep rule, but I don't think we need to commit this?